core/dbt/adapters/internal_storage/local_filesystem.py

```diff
@@ -1,557 +1,218 @@
-                                "searched_path": relative_path_to_search,
-                                "absolute_path": absolute_path,
-                                "relative_path": relative_path,
-                            }
-                        )
-
-        return matching
-
-    # switch to read
-    def load_file_contents(path: str, strip: bool = True) -> str:
-        # sys.stdout.write("Using local file adapter\n")
-        path = Adapter._convert_path(path)
-        with open(path, "rb") as handle:
-            to_return = handle.read().decode("utf-8")
-
-        if strip:
-            to_return = to_return.strip()
-
-        return to_return
-
-    # switch to write
-    def make_directory(path: str) -> None:
-        """
-        Make a directory and any intermediate directories that don't already
-        exist. This function handles the case where two threads try to create
-        a directory at once.
-        """
-        path = Adapter._convert_path(path)
-        if not os.path.exists(path):
-            # concurrent writes that try to create the same dir can fail
-            try:
-                os.makedirs(path)
-
-            except OSError as e:
-                if e.errno == errno.EEXIST:
-                    pass
-                else:
-                    raise e
-
-    # switch to write
-    def make_file(path: str, contents: str = "", overwrite: bool = False) -> bool:
-        """
-        Make a file at `path` assuming that the directory it resides in already
-        exists. The file is saved with contents `contents`
-        """
-        if overwrite or not os.path.exists(path):
-            path = Adapter._convert_path(path)
-            with open(path, "w") as fh:
-                fh.write(contents)
-            return True
+def ready_check() -> bool:
+    """Ensures the adapter is ready for use.
+
+    Returns:
+        `True` if the resource is ready to be used.
+        `False` if the resource is not ready to be used.
+
+    Raises:
+        TBD - TODO: How best to report back errors here?
+        It should never fail for a filesystem (unless we're using something very exotic),
+        but for databases this should be our primary source of troubleshooting information.
+
+    """
+    return True
+
+
+def read(
+    path: str,
+    strip: bool = True,
+) -> str:
+    """Reads the content of a file on the filesystem.
+
+    Args:
+        path: Full path of file to be read.
+        strip: Wether or not to strip whitespace.
+
+    Returns:
+        Content of the file
+
+    """
+    # create a concrete path object
+    path: Path = Path(path)
+
+    # read the file in as a string, or none if not found
+    file_content = path.read_text(encoding="utf-8")
+
+    # conditionally strip whitespace
+    file_content = file_content.strip() if strip else file_content
+
+    return file_content
+
+
+def write(
+    path: str,
+    content: Union[str, bytes, None],
+    overwrite: bool = False,
```

Comment:
I'm not sure that "overwrite" should be off by default. Generally when we write out a file we really do want to replace it if it already exists. I think this ended up here because of combining two formerly different methods. It's the reason that the partial parsing tests are failing, because the partial parse msgpack file isn't being written out. And I think that would be the case for a lot of other cases, but that tends not to be the sort of thing we cover in our tests.

Reply:
Ahh.. Good to know.  Yeah we can switch that up.